erts/emulator/ryu/README.ryu_update.md

```diff
@@ -0,0 +1,18 @@
+# How to update the Ryu version used by Erlang
+
+Last commit taken : 844864ac213bdbf1fb57e6f51c653b3d90af0937
+
+## The basic changes to the Ryu library
+
+To work with the Erlang VM, Ryu has been changed in three important ways:
+
+1. We only kept the bare minimum files needed to generate a double to string with the shortest algorithm, with the widest lookup table
+2. We deleted the code producing the final string, this is handled using a modified version of to_chars from the MS STL. <https://github.com/microsoft/STL/blob/8f4c8163775d665d80642044ce27c4bc696127ce/stl/inc/xcharconv_ryu.h#L1302>
+3. All other unneeded code has been deleted
+
+This is build with our own makefile.
+
+Some of the more minor difference:
+
+- the Zero case in common.h is changed to correspond to erlang fixed point version
+- the MS STL pointer check are not here. Erlang generate a 256 bytes buffer, we only need 30 maximum. Beware what this mean when refactoring.
```

Comment:
I don't understand what this means. What is the MS STL pointer check and why don't we need it (or something similar to it)? What does this mean for refactoring?

I think it would be useful if you could mark all the changes you have done to the code from Ulf Adams and Microsoft and comment on why they are done in the code. I think this would be helpful when/if someone needs to update the code later. You could start every comment with a prefix (e.g. " // CHANGE_FOR_ERLANG: We do not need the MS STL pointer check that was here because ....."). What do you think?

If you do that you can write something like the following in this file: All the changes to the code from ... and ... have been marked with a comment starting with "CHANGE_FOR_ERLANG".

This will also make it easier for me to review your changes to the code.

Reply:
Fair will try to do that. If there is code i dropped, do you want to get it commented ?

The reason i dropped it is because we pass a size controlled buffer that will always be bigger than needed right now.

Reply:
Sounds good. Yes, I think that could help maintainability as the person that maintains/update the code can then hopefully understand why certain code was dropped faster.